lib/puppet/provider/package/pkgin.rb

```diff
@@ -12,8 +12,8 @@
   def self.parse_pkgin_line(package)
 
     # e.g.
-    #   vim-7.2.446 =        Vim editor (vi clone) without GUI
-    match, name, version, status = *package.match(/(\S+)-(\S+)(?: (=|>|<))?\s+.+$/)
+    #   vim-7.2.446;Vim editor (vi clone) without GUI
+    match, name, version, status = *package.match(/(\S+)-(\S+?);(?:(=|>|<))?.+$/)
```

Comment:
Too soon, it seems it's breaking on `ruby18-puppet-0.25.5nb1`; the irony...

`$1 == ruby18`
`$2 == puppet-0.25.5nb1`

The hyphen in `ruby18-puppet` shouldn't be a breaking character.